pkg/restore/restore_crd.go

```diff
@@ -0,0 +1,237 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package restore
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/debug"
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var getCrName = `
+	kubectl -n %s get %s -o json | jq -r '.items[] | select(.metadata.deletionGracePeriodSeconds == 0) | .metadata.name'
+`
+
+var getCrYamlContent = `
+	kubectl -n %s get %s %s -oyaml > %s.yaml
+`
+
+var getK8sYamlContent = `
+	kubectl -n %s get %s -oyaml > %s
+`
+
+var getCrUID = `
+	kubectl -n %s get %s %s -o 'jsonpath={.metadata.uid}'
+`
+
+var removeFinalizers = `
+	kubectl -n %s patch %s/%s --type json --patch='[ { "op": "remove", "path": "/metadata/finalizers" } ]'
+`
+
+var recreateCr = `
+	kubectl create -f %s
+`
+
+func RestoreCrd(ctx context.Context, k8sclientset *k8sutil.Clientsets, operatorNamespace, clusterNamespace string, args []string) {
+	cr := args[0]
+	command := fmt.Sprintf(getCrName, clusterNamespace, cr)
+	crName := strings.TrimSpace(exec.ExecuteBashCommand(command))
+
+	if len(args) == 2 {
+		crName = args[1]
+	}
+
+	logging.Info("Restoring crd %s name %s", cr, crName)
+
+	logging.Info("Scaling down the operator to 0")
+	err := debug.SetDeploymentScale(ctx, k8sclientset.Kube, operatorNamespace, "rook-ceph-operator", 0)
+	if err != nil {
```

Comment:
If the operator is already scaled down, will this fail?

Reply:
actual because calling this restore command, we call `clientsets := GetClientsets(cmd.Context())` to check rook version and there we first check if rook operator pod is running or not. We do this for all command
```
Warning: rook version 'rook: v1.12.0-alpha.0.227.g3e248bf9a' is running a pre-release version of Rook.
```
This is the first line of any command operation. So,we are safe we don't need to think about case where rook operator pod is already down

Reply:
Do you mean the tool will fail early if the rook operator pod is not running before coming here?

Reply:
it will initially print the log
```
"waiting for pod with label %q in namespace %q to be running
```
and after 60 iteration it will fail

Reply:
So if the operator is not running, it blocks all plugin commands? Instead, how about if we just print a warning that the version could not be detected because the operator is down. Then the commands won't be blocked. In some disaster recovery scenarios, it will likely be common that the admin could scale down the operator while they are troubleshooting, before running these commands.